lib/grape/validations/types/custom_type_coercer.rb

```diff
@@ -160,10 +122,15 @@ def enforce_symbolized_keys(type, method)
           # Collections have all values processed individually
           if [Array, Set].include?(type)
             lambda do |val|
-              method.call(val).tap do |new_value|
-                new_value.map do |item|
-                  item.is_a?(Hash) ? symbolize_keys(item) : item
+              # if the value cannot be parsed, return InvalidValue
+              begin
+                method.call(val).tap do |new_val|
+                  new_val.map do |item|
+                    item.is_a?(Hash) ? symbolize_keys(item) : item
+                  end
                 end
+              rescue StandardError => _e
```

Comment:
This is too broad of a catch-all, I think should be a parser error only.

Reply:
We don't know which error might be raised by a custom type. Users define it. :slightly_smiling_face:

Reply:
That's why it's a problem. Catch all could be something you actually don't expect and there's no way to know if you catch all. We should not be catching all here at all or having an explicit contract with a coercer. If you want the latter, introduce an exception such as CoercionError and rescue that. Document that coercers should throw this or a child of this exception.

Reply:
You are right, this `begin/rescue` isn't required. Most likely I added to fix some broken tests, after refactoring, it became useless.